matrix/common/etl/dcp_zarr_store.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+"""Class for working with expression matrices serialized in the DCP."""
+
+import glob
+import json
+import os
+from collections import MutableMapping
+
+
+class DCPZarrStore(MutableMapping):
+    """
+    Zarr compatible interface to expression matrices stored in a DCP analysis bundle.
+    Parameters
+    ----------
+    bundle_dir: str
+        Local directory where a DCP analysis bundle is stored.
+    Methods
+    -------
+    Provides the MutableMapping interface, as required by zarr. But bad news: the DCP isn't
+    mutable, so mutations raise NotImplementedError with a helpful note.
+    """
+
+    # The DSS doesn't allow "directories", which really means it doesn't allow
+    # file keys with a '/' in the name since none of the storage backends has a
+    # directory concept anyway. So, we replace '/' with a different separator.
+    _separator_char = "!"
+
+    # There are possibly many files in the analysis bundle, but we only want to
+    # read those that are associated with the expression matrix. We can
+    # identify those via a pattern in the filename
+    _zarr_pattern = f"*.zarr{_separator_char}*"
+
+    def _transform_key(self, key):
+        return key.replace("/", self._separator_char)
+
+    def __init__(self, bundle_dir: str):
+        """
+        :param bundle_dir: path to local dir containing analysis bundle
+        """
+        self.bundle_dir = bundle_dir
+        self.cell_suspension_id = json.load(open(
+            os.path.join(self.bundle_dir, "cell_suspension_0.json")))['provenance']['document_id']
+        self.cache = {}
+
+    def __setitem__(self, key, value):
+        raise NotImplementedError("The HCA Data Storage System is read-only.")
```

Comment:
Just so I understand, this is not pointing at the DSS at all right? It's some local directory that's a result of like `hca download`?

Reply:
True (technically GET file). Updating message to "DCPZarrStore is read-only" since we don't have a use for writing currently.